oarepo_ui/resources/resource.py

```diff
@@ -148,29 +151,19 @@ def detail(self):
             component_key="search",
         )
 
+
```

Comment:
The following lines should be inside the extension - that is current_oarepo_ui.catalog would become cached_property.
Then, singleton_check would not be necessary, would it?